app/services/sign_up_service.rb

```diff
@@ -0,0 +1,59 @@
+class SignUpService
+
+  attr_accessor :user, :status, :message
+
+  def initialize(user)
+    @user = user
+    @status = status
+    @message = message
+  end
+
+  def signup(invitation_token, stripe_token)
+    token = stripe_token
+    if user.valid?
+      charge_card = charge(token, user)
+      if charge_card.successful?
+        user.save
+        AppMailer.delay.send_welcome_message(@user)
+        connect_user_and_inviter(user, invitation_token)
+        @status = :success
+        @message = 'You are successful Registed and Signed in'
+        self
+      else
+        @status = :warning
+        @message = charge_card.response.to_s
+        self
+      end
+    else
+      @status = :warning
+      @message = 'user info is not valid, please fullfil the column highlighted'
+      self
```

Comment:
You could refactor 3 `self`s to end of the method as each `if` branch ends with it.